library/std/src/sys/theseus/thread.rs

```diff
@@ -1,33 +1,49 @@
-use super::unsupported;
-use crate::ffi::CStr;
-use crate::io;
-use crate::num::NonZeroUsize;
-use crate::time::Duration;
+use super::{current_task, current_task_id, io_err};
+use crate::{ffi::CStr, io, num::NonZeroUsize, sys::unsupported, time::Duration};
+use libtheseus::{mem, stdio, task};
 
-pub struct Thread(!);
+pub struct Thread(task::JoinableTaskRef);
 
-pub const DEFAULT_MIN_STACK_SIZE: usize = 4096;
+pub const DEFAULT_MIN_STACK_SIZE: usize = mem::KERNEL_STACK_SIZE_IN_PAGES * mem::PAGE_SIZE;
 
 impl Thread {
     // unsafe: see thread::Builder::spawn_unchecked for safety requirements
-    pub unsafe fn new(_stack: usize, _p: Box<dyn FnOnce()>) -> io::Result<Thread> {
-        unsupported()
+    pub unsafe fn new(stack: usize, p: Box<dyn FnOnce()>) -> io::Result<Thread> {
```

Comment:
is `stack` the size of the stack in bytes? if so, we should rename it as such. Otherwise one might interpret it as the intended virtual address of the stack